ndcube/ndcube.py

```diff
@@ -542,34 +539,24 @@ def explode_along_axis(self, axis):
         result : `ndcube_sequence.NDCubeSequence`
 
         """
-        # If axis is -ve then calculate the axis from the length of the dimensions of one cube
+        # is axis is -ve then calculate the axis from the length of the dimensions of one cube
         if axis < 0:
             axis = len(self.dimensions) + axis
-        # the range of the axis that needs to be sliced
-        range_of_axis = self.data.shape[axis]
         # To store the resultant cube
         result_cubes = []
-        # We need to convert extra_coords dictionnary to a liste of tuple to create a new NDCube
-        extra_coords_formated = utils.cube.convert_extra_coords_dict_to_input_format(
-            self.extra_coords, missing_axis=[False, False, True])
-        # Creating a (N-1)Dcube list
-        for i in range(range_of_axis):
-            cube = self._new_instance(
-                self[i].data, self[i].wcs, uncertainty=self[i].uncertainty, unit=self.unit,
-                meta=self.meta, mask=self[i].mask, missing_axis=[False, False, True],
-                extra_coords=extra_coords_formated)
-            result_cubes.append(cube)
-        return NDCubeSequence(result_cubes, meta=self.meta)
-
-    @classmethod
-    def _new_instance(cls, data, wcs, uncertainty, unit, meta,
-                      mask, extra_coords, missing_axis=None):
-        """
-        Instantiate a new instance of this class using given data.
-        """
-        return cls(data=data, wcs=wcs, uncertainty=uncertainty, unit=unit,
-                   meta=meta, mask=mask, extra_coords=extra_coords,
-                   missing_axis=missing_axis)
+        # All slices are initially initialised as slice(None, None, None)
+        cube_slices = [slice(None, None, None)] * len(self.data.ndim)
+        # Slicing the cube inside result_cube
+        for i in range(self.data.shape[axis]):
+            # Setting the slice value to the index so that the slices are done correctly.
+            cube_slices[axis] = i
+            # Appending the sliced cubes in the result_cube list
+            result_cubes.append(self[cube_slices])
```

Comment:
The way this is arranged now, the metadata of the Cube will be copied for each slice.  Really, the metadata still applies to all the data and isn't specific to the slices.  So I think the metadata should be moved from the cube level, to the sequence level.  This could simply be done by:
```python
    sliced_cube = self[cube_slices]
    sliced_cube.meta = None
    result_cubes.append(self[cube_slices])
return NDCubeSequence(result_cubes, common_axis=axis, meta=self.meta)
```